src/runtime/int.cpp

```diff
@@ -1157,30 +1212,64 @@ void setupInt() {
                                                                                    ParamNames::empty(), CAPI)));
 
     _addFuncIntFloatUnknown("__add__", (void*)intAddInt, (void*)intAddFloat, (void*)intAdd);
```

Comment:
Looks like the issue is that int no longer would know how to handle `int.__add__(float)`, and we don't (yet) rewrite cases where we have to fall back to trying `__radd__`.